cypress/support/index.d.ts

```diff
@@ -0,0 +1,16 @@
+/// <reference types="cypress" />
+
+declare namespace Cypress {
+  interface Chainable {
+    /**
+     * Custom command to select DOM element by data-cy attribute.
+     * @example cy.dataCy('greeting')
+     */
+    login(): Chainable<Element>;
+    preserveSession(): Chainable<Element>;
+    setPrivacyTogglesToPublic(): Chainable<Element>;
+    goToSettings(): Chainable<Element>;
+    typeUsername(username: string): Chainable<Element>;
+    resetUsername(): Chainable<Element>;
+  }
+}
```

Comment:
I'm not sure why the Cypress docs recommend this approach, but there's no need for you to figure out the types - you can infer them.

For example, you can put the declaration inside `commands.ts` and write things like this

```ts
const login = () => {
  cy.visit(`${Cypress.env('API_LOCATION')}/signin`);
  cy.contains('Welcome back');
}

Cypress.Commands.add('login', login);

declare global {
  namespace Cypress {
    interface Chainable {
      login: typeof login;
    }
  }
}
```

Reply:
So, this file can go.